cmds/coredhcp/config.yml.example

```diff
@@ -9,5 +9,7 @@ server6:
 server4:
     listen: '0.0.0.0:67'
     interface: "eth0"
-    plugins:
-        - server_id: 192.168.1.12
+    plugins:        
+        #- server_id: 10.10.10.1
+        #- file: leases.txt 10.10.10.1/24 60
```

Comment:
After giving it a though, I am not sure this is the proper format we should use to define a pool of IPs.
For instance, if you define `10.10.10.1/24` the DHCP server may assign a client the IP `10.10.10.1` which you may want to use as a static IP for a router.

I think that instead this should have a starting IP and an ending IP for the pool. Ex. `10.10.10.100 10.10.10.200`.

In that way you can avoid having to use the whole subnet for dynamic IPs. Thoughts?

Reply:
And we can have the `netmask` as another plugin/option.

Reply:
There was a check so the address of the router is never leased, but now I think it is not doing what it's supposed to do because the serverIP is not the same as the router IP. Yes it can be implemented so a range of IPs is set. Do you think file plugin arguments should look like:
`file: leases.txt 10.10.10.1/24 10.10.10.100 10.10.10.200  60`
or
`file: leases.txt 10.10.10.1 10.10.10.100 10.10.10.200  60`

Reply:
I think it should be either like this

```
file: leases.txt 10.10.10.100 10.10.10.200 60
```

There is no need to add `10.10.10.1/24` or `10.10.10.1`.

Or even like this:

```
file: leases.txt 10.10.10.100 10.10.10.200
```

I wonder if we can decouple the lease time as well.

Reply:
> I wonder if we can decouple the lease time as well.

How would the file plugin save the leases with the time of expiry?
> ```
> file: leases.txt 10.10.10.100 10.10.10.200
> ```
How would I get the serverIP from here?

Reply:
> How would I get the serverIP from here?

What do you need a `serverIP` for in the `file` plugin?

Reply:
I don't need the server IP only if I start at the bottom range and continue to increment the IP until a free lease is found. The way it's implemented now is a random IP (which can be later a hash function so the same device always gets the same initial IP ), and then it is adjusted so it fits the netmask. To do that it uses the server IP and the server netmask.
`for i := 0; i < 4; i++ {`
		`ip[i] = (ip[i] & (network.Mask[i] ^ 255)) | (network.IP[i] & network.Mask[i])`
`}`

Reply:
If you want to randomize you can still do it, you can pick an IP from the pool. You only need to use a different logic.